### PR TITLE
Google event decorator

### PR DESCRIPTION
**Pull request by jtkpiotr**
@adamzima @s1mplex @michal-konarski @lkudyba @quirk0o @mhib

### CODE REVIEW

File: app/models/google_calendar/event_wrapper/builder.rb [hunk 1]
```diff
@@ -1,8 +1,89 @@
...
+      end
+
+      def build_attendees(attendees)
+        result = attendees.nil? ? [] : attendees.map { |attendee| new_attendee(attendee) }
+        result + [room_attendee, creator_attendee].compact
+      end
+
+      def room_attendee
+        return unless conference_room
+        new_attendee email: conference_room.email, response_status: 'accepted'
```
Comment (jtkpiotr) [inline app/models/google_calendar/event_wrapper/builder.rb]: const?

File: app/models/google_calendar/event_wrapper/event.rb [hunk 2]
```diff
@@ -1,73 +1,36 @@
...
-
-      def initialize(**params)
-        @id = params[:id]
-        @summary = params[:summary]
-        @description = params[:description]
-        @current_user_email = params[:user_email]
-        @attendees = params[:attendees]
+    class Event < DelegateClass(::Google::Apis::CalendarV3::Event)
+      attr_accessor :conference_room, :user_email, :google_event
+      FIELDS = %i{id start end description creator attendees summary}
```
Comment (jtkpiotr) [inline app/models/google_calendar/event_wrapper/event.rb]: use dots

File: app/models/google_calendar/event_wrapper/event.rb [hunk 3]
```diff
@@ -1,73 +1,36 @@
...
-        attendees + [room_attendee, creator_attendee].compact
+        return false unless start.present? || end_time.present?
+        start.date_time.present? && end_time.date_time.present?
       end
 
-      def room_attendee
-        return unless conference_room
-        Google::Apis::CalendarV3::EventAttendee.new(email: conference_room.email, response_status: 'accepted')
+      def end_time
+        send(:end)
```
Comment (jtkpiotr) [inline app/models/google_calendar/event_wrapper/event.rb]: <3 so sexy

File: app/models/google_calendar/event_wrapper/event.rb [hunk 4]
```diff
@@ -1,73 +1,40 @@
...
-          id: id,
-          start: start_time.to_h,
-          end: end_time.to_h,
-          description: description,
-          conference_room: conference_room,
-          creator: creator.to_h,
-          attendees: attendees.map(&:to_h),
-          summary: summary
-        }
+        super.slice(*FIELDS).merge conference_room: conference_room
```
Comment (mjk-old) [inline app/models/google_calendar/event_wrapper/event.rb]: ()

File: app/models/google_calendar/event_wrapper/rounded_event.rb [hunk 5]
```diff
@@ -1,11 +1,10 @@
 module GoogleCalendar
   module EventWrapper
     class RoundedEvent < GoogleCalendar::EventWrapper::Event
-      attr_accessor :start_timestamp, :end_timestamp, :all_day
+      attr_accessor :start_timestamp, :end_timestamp
 
-      def initialize(**params)
+      def initialize(google_event, params = {})
         super
```
Comment (michal-konarski) [inline app/models/google_calendar/event_wrapper/rounded_event.rb]: `super(google_event)`?

**Comment by mjk-old** [comment]
🆗

**Pull request merged**